homework2/hw1.py

```diff
@@ -0,0 +1,68 @@
+"""
+Given a file containing text. Complete using only default collections:
+    1) Find 10 longest words consisting from largest amount of unique symbols
+    2) Find rarest symbol for document
+    3) Count every punctuation char
+    4) Count every non ascii char
+    5) Find most common non ascii char for document
+"""
+import string
+from collections import Counter
+from typing import List
+
+
+def get_longest_diverse_words(file_path: str) -> List[str]:
+    with open(file_path) as file:
+        lines = []
+        for line in file:
+            lines += line.strip().split()
+
+    return sorted(lines, key=lambda x: len(x) and len(Counter(x)), reverse=True)[:10]
```

Comment:
Why exactly you use `Counter` here? Is there any chance that the same word will be in top-10 twice?

Reply:
Also your key is not working as you intended. It returns counter object.

Reply:
As I understand we should find strings with maximum length and maximum unique symbols, that's why I decided to use Counter()

Reply:
About using two keys I saw an example [here](https://www.kite.com/python/answers/how-to-sort-by-two-keys-in-python)

Reply:
Also in file data.txt I saw a lot of Unicode symbols such as '\\u00f6' but the task sounds just _"Find 10 longest words consisting from largest amount of unique symbols"_, should I consider Unicode symbols or only ASCII letters?

Reply:
Also, I see punctuation symbols in the data.txt, which symbols I should consist of? I mean that exist a lot of words with "-" or " ' " inside

Reply:
> As I understand we should find strings with maximum length and maximum unique symbols, that's why I decided to use Counter()

`len(Counter(x))` is equivalent to `len(set(x))` here, so you did not need `Counter`.

> About using two keys I saw an example [here](https://www.kite.com/python/answers/how-to-sort-by-two-keys-in-python)

The example uses `tuple` to compare the strings, but you have a `bool` key here.

> Also in file data.txt I saw a lot of Unicode symbols such as '\u00f6' but the task sounds just _"Find 10 longest words consisting from largest amount of unique symbols"_, should I consider Unicode symbols or only ASCII letters?

You should count unicode characters as one character, so `\u00f6` should be `ö`. You need right encoding to be able to do it.

> Also, I see punctuation symbols in the data.txt, which symbols I should consist of? I mean that exist a lot of words with "-" or " ' " inside

This depends on the tokenization method you use, so there is no definite answer here. `str.split` is valid tokenizer for the task.